prow/gerrit/client/client.go

```diff
@@ -252,6 +254,37 @@ func (c *Client) Authenticate(cookiefilePath, tokenPath string) {
 	}
 }
 
+// UpdateClients update gerrit clients with new instances map
+func (c *Client) UpdateClients(instances map[string][]string) error {
+	c.lock.Lock()
```

Comment:
What exactly does this lock protect? It seems like it protects the `authentication` field and some fields of the handlers?

The `handlers` map is used without holding the lock elsewhere, but that seems no longer safe.

Reply:
As I understand this lock protects `handlers` and `accounts` map, as well as `authentication` function.

Agree that this is no longer thread safe, updated all usage of handlers to be thread safe